lib/improver/tests/nowcasting/pysteps_advection/test_PystepsExtrapolate.py

```diff
@@ -0,0 +1,231 @@
+    rain_cube.remove_coord('forecast_reference_time')
+
+    return rain_cube
+
+
+def _make_orogenh_cube(time_now, interval, max_lead_time):
+    """Construct an orographic enhancement cube with data valid for
+    every lead time"""
+    orogenh_data = 0.05*np.ones((8, 8), dtype=np.float32)
+    orogenh_cube = set_up_variable_cube(
+        orogenh_data, name='orographic_enhancement', units='mm/h',
+        spatial_grid='equalarea', time=time_now, frt=time_now)
+
+    time_points = [time_now]
+    lead_time = 0
+    while lead_time <= max_lead_time:
+        lead_time += interval
+        new_point = (
+            time_points[-1] + datetime.timedelta(seconds=60*interval))
+        time_points.append(new_point)
+
+    orogenh_cube = add_coordinate(
+        orogenh_cube, time_points, "time", is_datetime=True)
+    return orogenh_cube
+
+
+class Test_process(IrisTest):
+    """Test wrapper for pysteps semi-Lagrangian extrapolation"""
+
+    def setUp(self):
+        """Set up test velocity and rainfall cubes"""
+        time_now = datetime.datetime(2019, 9, 10, 15)
+        wind_data = 4*np.ones((8, 8), dtype=np.float32)
+        self.ucube = set_up_variable_cube(
+            wind_data, name='precipitation_advection_x_velocity',
+            units='m/s', spatial_grid='equalarea', time=time_now,
+            frt=time_now)
+        self.vcube = set_up_variable_cube(
+            wind_data, name='precipitation_advection_y_velocity',
+            units='m/s', spatial_grid='equalarea', time=time_now,
+            frt=time_now)
+        self.rain_cube = _make_initial_rain_cube(time_now)
+
+        self.interval = 15
+        self.max_lead_time = 120
+        self.orogenh_cube = _make_orogenh_cube(
+            time_now, self.interval, self.max_lead_time)
+
+        # set up all grids with 3.6 km spacing (1 m/s = 3.6 km/h,
+        # using a 15 minute time step this is one grid square per step)
+        xmin = 0
+        ymin = 200000
+        step = 3600
+        xpoints = np.arange(xmin, xmin+8*step, step).astype(np.float32)
+        ypoints = np.arange(ymin, ymin+8*step, step).astype(np.float32)
+        for cube in [
+                self.ucube, self.vcube, self.rain_cube, self.orogenh_cube]:
+            cube.coord(axis='x').points = xpoints
+            cube.coord(axis='y').points = ypoints
+
+    def test_basic(self):
+        """Test output is a list of cubes with expected contents and
+        global attributes"""
+        result = PystepsExtrapolate().process(
+            self.rain_cube, self.ucube, self.vcube, self.interval,
+            self.max_lead_time, self.orogenh_cube)
+        self.assertIsInstance(result, list)
+        # check result is a list including a cube at the analysis time
+        self.assertEqual(len(result), 9)
+        self.assertIsInstance(result[0], iris.cube.Cube)
+        self.assertIsInstance(result[0].data, np.ma.MaskedArray)
+        self.assertEqual(result[0].data.dtype, np.float32)
```

Comment:
Doesn't actually test that the data is unchanged in the analysis time cube.

Reply:
Added something to check this.